packages/labs/context/src/lib/decorators/provide.ts

```diff
@@ -48,15 +47,41 @@ export function provide<ValueType>({
 }: {
   context: Context<unknown, ValueType>;
 }): ProvideDecorator<ValueType> {
-  return decorateProperty({
-    finisher: (ctor: typeof ReactiveElement, name: PropertyKey) => {
-      const controllerMap = new WeakMap();
-      ctor.addInitializer((element: ReactiveElement): void => {
-        controllerMap.set(element, new ContextProvider(element, {context}));
+  return (<C extends ReactiveElement, V extends ValueType>(
+    protoOrTarget: ClassAccessorDecoratorTarget<C, V>,
+    nameOrContext: PropertyKey | ClassAccessorDecoratorContext<C, V>
+  ) => {
+    // Map of instances to controllers
+    const controllerMap = new WeakMap();
+    if (typeof nameOrContext === 'object') {
+      nameOrContext.addInitializer(function (this: C) {
```

Comment:
Here and for consume: let's add a comment saying essentially "this is the standard decorators branch" and "this is the legacy experimental decorators branch"

Reply:
done